jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/UserTaskServiceImpl.java

```diff
@@ -188,6 +188,8 @@ public void claim(String deploymentId, Long taskId, String userId) {
 			// perform actual operation
 			taskService.claim(taskId, userId);
 		} catch(PermissionDeniedException e){
+		    throw new PermissionDeniedException(e.getMessage());
+        } catch(TaskNotFoundException e){
             throw new TaskNotFoundException(e.getMessage());
```

Comment:
The part
```
} catch(PermissionDeniedException e){
    throw new PermissionDeniedException(e.getMessage());
} catch(TaskNotFoundException e){
    throw new TaskNotFoundException(e.getMessage());
```
can be completely omitted I believe. Since you catch the same type of exception and then throwing it again with the same message, there is no reason to do it, Java will do it automatically. So I think we can delete these 4 lines completely and just leave try {...} finally {...} block.

Reply:
the change itself is sort of needed as currently the PermissionDeniedException is caught and rethrown is TaskNotFound so both needs to be now handled separately ... but indeed, we might not need the catch at all...